Src/PerformanceCollector/Shared/QuickPulseTelemetryProcessor.cs

```diff
@@ -178,55 +184,96 @@ private static ITelemetryDocument ConvertRequestToTelemetryDocument(RequestTelem
                            DependencyTypeName = dependencyTelemetry.DependencyTypeName,
-                           DependencyKind = dependencyTelemetry.DependencyKind
+                           DependencyKind = dependencyTelemetry.DependencyKind,
+                           Properties = GetProperties(dependencyTelemetry, SpecialDependencyPropertyName)
                        };
         }
 
         private static ITelemetryDocument ConvertExceptionToTelemetryDocument(ExceptionTelemetry exceptionTelemetry)
         {
-            // //!!! cut length for Exception.ToString()
             return new ExceptionTelemetryDocument()
                        {
+                           Id = Guid.NewGuid(),
                            Version = TelemetryDocumentContractVersion,
-                           Message = exceptionTelemetry.Message,
+                           Message = TruncateValue(exceptionTelemetry.Message),
                            SeverityLevel =
                                exceptionTelemetry.SeverityLevel != null
                                    ? exceptionTelemetry.SeverityLevel.Value.ToString()
                                    : null,
                            HandledAt = exceptionTelemetry.HandledAt.ToString(),
                            Exception =
-                               exceptionTelemetry.Exception != null ? exceptionTelemetry.Exception.ToString() : null
+                               exceptionTelemetry.Exception != null
+                                   ? TruncateValue(exceptionTelemetry.Exception.ToString())
+                                   : null,
+                           ExceptionType =
+                               exceptionTelemetry.Exception != null
+                                   ? TruncateValue(exceptionTelemetry.Exception.GetType().FullName)
+                                   : null,
+                           ExceptionMessage =
+                               exceptionTelemetry.Exception != null
+                                   ? TruncateValue(exceptionTelemetry.Exception.Message)
+                                   : null,
+                           Properties = GetProperties(exceptionTelemetry)
                        };
         }
 
+        private static KeyValuePair<string, string>[] GetProperties(ISupportProperties telemetry, string specialPropertyName = null)
+        {
+            Dictionary<string, string> properties = null;
+
+            if (telemetry.Properties != null && telemetry.Properties.Count > 0)
+            {
+                properties = new Dictionary<string, string>(MaxPropertyCount + 1);
+
+                foreach (var prop in
+                    telemetry.Properties
+                    .Where(p => !string.Equals(p.Key, specialPropertyName, StringComparison.Ordinal))
+                    .Take(MaxPropertyCount))
+                {
+                    properties.Add(TruncateValue(prop.Key), TruncateValue(prop.Value));
```

Comment:
Are keys really allowed to be 32768 characters long?  That seems excessive for a key in my opinion.